src/core/Akka.Persistence.TCK.Tests/MemorySnapshotStoreSpec.cs

```diff
@@ -23,5 +32,129 @@ public MemorySnapshotStoreSpec(ITestOutputHelper output)
             Initialize();
         }
 
+        [Fact]
+        public void MemorySnapshotStore_is_threadsafe()
+        {
+            EventFilter.Error().Expect(0, () =>
```

Comment:
This comments describe most of what's going on inside this spec, but the gist of it is: spin up several `SnapshotStore` calls from persistent actors and force it to barf. The cause of the original `InvalidOperationException` is because the collection was being modified concurrently from multiple threads.